### ISSUE TITLE
cannot copy or move to sd card mount unless sudo lf

### ISSUE THREAD

**Issue by windowswithoutborders** [gokcehan/lf#415]
Hi everyone,

tl;dr I cannot perform copy/move actions to my sd card mount within lf unless I launch it through sudo. Is there a way I can accomplish these tasks without having to sudo into lf?

To preface, I am new to linux and have been searching google and looking through this forum to see if I could find a solution to this problem, but have come up empty. I have a feeling/hope that this issue is actually easy to resolve and it's just something i'm overlooking as a novice.

I am on archlinux on the latest lts kernal, and I'm running the latest built-from-source binary at the time of this posting. I updated after seeing a post regarding moving files onto another drive causing "invalid cross-device link" hoping my issue might be related. Additionally, I can perform copy/move tasks within lf on my local file system without issue.

I mount my sd card with the following command: 
`_ mount -t exfat -o rw,gid=users,fmask=117,dmask=007 /dev/sdc1 ~/media/sdcard`

The sd card successfully mounts and appears as any other folder on my system, and from the command line itself, without any sudo commands, I can copy and move files to the sd card successfully. Unfortunately when I'm within LF copy and move commands result in:
`[1] copy: chmod ../media/sdcard/file.zip: operation not permitted` 

This issue is resolved by doing sudo lf, but ideally I can do the same without having to do that. I feel like this is probably easy and most likely unrelated to lf itself, and I'm just overlooking something. I greatly appreciate any assistance. Thanks.

**Comment by gokcehan** [comment]
@windowswithoutborders After we copy the file, we try to set the mod of the new file to the mod of the original file to preserve it [here](https://github.com/gokcehan/lf/blob/master/copy.go#L73), which seems to be failing in your case, likely due to fmask/dmask options given when mounting. Does `cp --preserve=mode` from shell work on these files? Can you try removing fmask/dmask options to see if it solves the issue?

I'm not sure we should consider this a bug, but still I will mark this as such for visibility in the meantime.

**Comment by windowswithoutborders** [comment]
> @windowswithoutborders After we copy the file, we try to set the mod of the new file to the mod of the original file to preserve it [here](https://github.com/gokcehan/lf/blob/master/copy.go#L73), which seems to be failing in your case, likely due to fmask/dmask options given when mounting. Does `cp --preserve=mode` from shell work on these files? Can you try removing fmask/dmask options to see if it solves the issue?
> 
> I'm not sure we should consider this a bug, but still I will mark this as such for visibility in the meantime.

Hi gokcehan,

Greatly appreciate your reply. Thank you for suggesting to try `cp --preserve=mode`. Indeed, with this option added I received an error in my shell `cp: failed to preserve ownership`, and removing the fmask/dmask options as well as just trying a plethora of additional options when mounting did not resolve my issue. Long story short I ended up going on a very long and winding google adventure spurred from googling the error result I saw in my shell that lead me to an alternate mounting program called **udevil**. After dedicating some additional time figuring out why that was giving me difficulty with just mounting my exfat sd card, I found the solution to that was [simply creating a new directory in /media and mounting to it.](https://bugs.gentoo.org/704728#c2) Mounting with udevil did in fact solve everything, and I am now able to copy and move to my sd card through lf flawlessly. Thanks again, gokcehan, as I probably wouldn't have figured this out without your suggestion, lf is absolutely amazing and I am happy to be getting the functionality I want out of it :)!
